mpas_analysis/test/test_generalized_reader.py

```diff
@@ -0,0 +1,100 @@
+                                        calendar=calendar,
+                                        time_variable_name='Time',
+                                        variable_list=var_list,
+                                        variable_map=variable_map,
+                                        year_offset=1850)
+
+            # make sure the remapping happened as expected
+            self.assertEqual(sorted(ds.data_vars.keys()), sorted(var_list))
+
+    def test_open_dataset_fn(self):
+        file_name = str(self.datadir.join('example_jan.nc'))
+        timestr = ['xtime_start', 'xtime_end']
+        var_list = ['time_avg_avgValueWithinOceanRegion_avgSurfaceTemperature']
+
+        for calendar in ['gregorian', 'gregorian_noleap']:
+            ds = open_multifile_dataset(file_names=file_name,
+                                        calendar=calendar,
+                                        time_variable_name=timestr,
+                                        variable_list=var_list,
+                                        year_offset=1850)
+            self.assertEqual(ds.data_vars.keys(), var_list)
+
+    def test_start_end(self):
+        file_name = str(self.datadir.join('example_jan_feb.nc'))
+        timestr = ['xtime_start', 'xtime_end']
+        var_list = ['time_avg_avgValueWithinOceanRegion_avgSurfaceTemperature']
+
+        for calendar in ['gregorian', 'gregorian_noleap']:
+            # all dates
+            ds = open_multifile_dataset(file_names=file_name,
+                                        calendar=calendar,
+                                        time_variable_name=timestr,
+                                        variable_list=var_list,
+                                        start_date='0001-01-01',
+                                        end_date='9999-12-31',
+                                        year_offset=1850)
+            self.assertEqual(len(ds.Time), 2)
+
+            # just the first date
+            ds = open_multifile_dataset(file_names=file_name,
+                                        calendar=calendar,
+                                        time_variable_name=timestr,
+                                        variable_list=var_list,
+                                        start_date='0005-01-01',
+                                        end_date='0005-02-01',
+                                        year_offset=1850)
+            self.assertEqual(len(ds.Time), 1)
+
+            # just the second date
+            ds = open_multifile_dataset(file_names=file_name,
+                                        calendar=calendar,
+                                        time_variable_name=timestr,
+                                        variable_list=var_list,
+                                        start_date='0005-02-01',
+                                        end_date='0005-03-01',
+                                        year_offset=1850)
+            self.assertEqual(len(ds.Time), 1)
+
```

Comment:
This is really great @xylar -- testing of opening file will hopefully ensure that we have all the functionality we need and if libraries change and this breaks we should know!